File: seminars/Sem07_OOP_Desing_and_SOLID_part_2/OnlineTasks/src/Application/Entities/Coffee.java
package Application.Entities;

import Application.Abstracts.HotDrink;

public class Coffee extends HotDrink {

    private String brand;
    private String sort;
    private boolean isRoasted;

    public Coffee(int id, int temperature, float coast, String brand, String sort, boolean isRoasted) {
        super(id, temperature, coast);
        setBrand(brand);
        setSort(sort);
        setRoasted(isRoasted);
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getSort() {
        return sort;
    }

    public void setSort(String sort) {
        this.sort = sort;
    }

    public boolean isRoasted() {
        return isRoasted;
    }

    public void setRoasted(boolean roasted) {
        isRoasted = roasted;
    }

    @Override
    public String toString() {
        return "Coffee{" +
                ", id=" + getId() +
                ", temperature=" + getTemperature() +
                ", coast=" + getCoast() +
                ", brand='" + getBrand() + '\'' +
                ", sort='" + getSort() + '\'' +
                ", isRoasted=" + isRoasted() +
                '}';
    }
}

File: seminars/Sem07_OOP_Desing_and_SOLID_part_2/OnlineTasks/FromJavaToPython/Model/Coffee.py
from .HotDrink import HotDrink


class Coffee(HotDrink):

    def __init__(self, my_id, temperature, cost, is_ready_to_serve, brand, sort):
        super().__init__(my_id, temperature, cost, is_ready_to_serve)
        self.brand = brand
        self.sort = sort

    def get_id(self):
        return self.id

    def __str__(self) -> str:
        return (f'Coffee [id={self.id}'
                f', temperature={self.temperature}'
                f', cost={self.cost}'
                f', is_ready_to_serve={self.is_ready_to_serve}'
                f', brand={self.brand}'
                f', sort={self.sort}]')
